### DISCUSSION TITLE
Plans for recursive import from a repo

### DISCUSSION THREAD

**Discussion by prodigysml** [PortSwigger/BChecks#78] [General]
Hey team! Firstly, the bchecks feature is taking off well and working a charm! I was curious if there were any plans to have a recursive import from a directory, or repo, within the burp suite tool itself (maybe even an auto-update / pull feature). Are any if those on the books yet? If not, another alternative could be to write an action that clones all the current checks and makes a single directory with them all, then does a zip and release. Happy to write that if the tool won't be supporting either of the above soon.

**Comment by Hannah-PortSwigger** [comment]
Hi.

We would like to add a "BChecks Store" to Burp in the future, potentially similar to how the BApp Store is currently, to make it easy to find and add BChecks in-product.

I have an extension that retrieves all BChecks from specified GitHub folders and saves them as individual files in a folder - you can find this [here](https://github.com/Hannah-PortSwigger/SaveBchecksToFile). It does have several improvements to make, but you could use this as a base for a different extension, or submit a pull request if there are any changes you wish to make.

**Reply by prodigysml** [reply]
That's awesome! Thanks for that! I'll try see if I can contribute there!
